specification/eventgrid/data-plane/Microsoft.AVS/preview/2023-09-01-preview/PrivateCloud.json

```diff
@@ -0,0 +1,216 @@
+        }
+      ],
+      "properties": {}
+    },
+    "AVSClusterFailedEventData": {
+      "description": "Schema of the Data property of an EventGridEvent for a Microsoft.AVS.ClusterFailed event.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/AVSClusterEventProperties"
+        }
+      ],
+      "properties": {
+        "failureMessage": {
+          "description": "Failure reason of an event.",
+          "type": "string"
+        }
+      }
+    },
+    "AVSScriptExecutionStartedEventData": {
+      "description": "Schema of the Data property of an EventGridEvent for a Microsoft.AVS.ScriptExecutionStarted event.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/AVSScriptExecutionEventProperties"
+        }
+      ],
+      "properties": {}
+    },
+    "AVSScriptExecutionFinishedEventData": {
+      "description": "Schema of the Data property of an EventGridEvent for a Microsoft.AVS.ScriptExecutionFinished event.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/AVSScriptExecutionEventProperties"
+        }
+      ],
+      "properties": {
+        "namedOutputs": {
+          "description": "Named outputs of completed execution, if any.",
+          "type": "object",
+          "items": {
+            "type": "string"
+          }
+        }
+      }
+    },
+    "AVSScriptExecutionCancelledEventData": {
+      "description": "Schema of the Data property of an EventGridEvent for a Microsoft.AVS.ScriptExecutionCancelled event.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/AVSScriptExecutionEventProperties"
+        }
+      ],
+      "properties": {}
+    },
+    "AVSScriptExecutionFailedEventData": {
+      "description": "Schema of the Data property of an EventGridEvent for a Microsoft.AVS.ScriptExecutionFailed event.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/AVSScriptExecutionEventProperties"
+        }
+      ],
+      "properties": {
+        "failureMessage": {
+          "description": "Failure reason of an event.",
+          "type": "string"
+        }
+      }
+    },
+    "AVSPrivateCloudEventProperties": {
+      "description": "Schema of the Data property of an EventGridEvent for Microsoft.AVS/privateClouds events.",
+      "type": "object",
+      "properties": {
+        "operationId": {
+          "description": "Id of the operation that caused this event.",
+          "type": "string"
+        }
+      }
+    },
+    "AVSClusterEventProperties": {
+      "description": "Schema of the Data property of an EventGridEvent for Microsoft.AVS/clusters events.",
+      "type": "object",
+      "properties": {
+        "operationId": {
+          "description": "Id of the operation that caused this event.",
+          "type": "string"
+        },
+        "hostsAdded": {
```

Comment:
we've had some back and forth about the naming here, I'd rather not use "name", as in "hostNamesAdded" because we are not adding names, we're adding hosts. Normally I'd model this with the type system, defining `FQDN` to use instead of string, regardless... imo anyone using this event will not be under any illusion that this is a collection of hosts.

Reply:
I think "addedHostNames" addresses the confusion around adding hosts vs names.

Reply:
I don't believe it does address the confusion, as "hostNames" is not what is affected by the event, but the cluster's hosts composition. The "name" itself is not sufficiently precise either and potentially bigger source of ambiguity here, but I'm sure we all have bigger fish to fry than argue about this.